### ISSUE TITLE
Screenreaders have no context when entering the survey or interacting with the control button

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#5834]
a11y severity: high

The first thing screenreaders hit is the “minimize” button without context on what they’ve gotten themselves into.

It would be ideal to use an `aria-labelledby` attribute on the survey container to reference the survey header ID, as that is the most proper way to approach this, but the ID generated on the header appears to be dynamic (docs-internal-guide-#####-####-morenonsense). If it’s possible to make this static in SurveyGizmo, let’s do that, and use that ID in the `aria-labelledby`. Similarly, we should have an `aria-describedby` ID for the “content” of the survey, which currently has a class of “sg-question-options” and does not have an ID. If we can change both of these IDs and reference the values, let’s use that approach.

If we are unable to make the above changes, as an alternate and easier option, the entire survey container can have an `aria-label` that reads something like, “Reset password optional survey,” but we don’t want to be overly verbose as we only want to communicate to someone what they’re about to be interacting with if they can’t see what they’ve just tabbed into. Additionally, when the screenreader moves past the minimize/expand button, the user is read “Password Reset - 1” when they reach the iframe. If this can instead be found and changed in Survey Gizmo to “Password reset optional survey,” (or just “survey,” but “optional” just gives them more context) then the `aria-label` could be simplified even further with just “survey”. This will come together and make more sense when it’s worked through and gone through with a screenreader and sensitivity to a user that may be blind. Does what’s read make sense - would you understand what’s going on without any visuals or overall picture? It’s a little subjective and there are likely many options that will be acceptable. The screenreader user should be able to tab into the survey and have this context read to them before hitting the minimize/expand button.

When the survey is expanded:

* The minimize/expand button should have an `aria-label` that communicates what interacting with the button with will do such as “minimize survey” or “hide survey”
* The minimize/expand button should have an `aria-expanded=“true”` on it

When the survey is minimized:

* The survey content should have an `aria-hidden=“true”` on it - relates to FXA-2218, if `display: none` works then we don't need this
* The minimize/expand button should have an `aria-label` that communicates what interacting with the button with will do such as “expand survey” or “show survey”
* The minimize/expand button should have an `aria-expanded=“false” on it

┆Issue is synchronized with this [Jira Bug](https://jira.mozilla.com/browse/FXA-2221)
┆Issue Number: FXA-2221

**Comment by data-sync-user** [comment]
➤ Lauren Zugai commented:

Note: this issue can be started without product, as all the bullet points above are still applicable, but we need product on the header/content IDs as I'm not sure if we can add those or not through SurveyGizmo.

**Comment by data-sync-user** [comment]
➤ John Gruen commented:

Here's what i think we should do...

  1. Wrap the survey in an aria label "Firefox accounts optional user survey"
  2. Change the minimize icon to a dismiss icon and fully remove the survey from the dom if a user clicks it
  3. Have the dismiss icon have an aria label of "dismiss survey" or similar

**Comment by data-sync-user** [comment]
➤ Alex Davis commented:

I personally would prefer to stick to minimizing rather than dismissing since the minimize allows people to complete what they're doing and then they can answer the survey.

**Comment by data-sync-user** [comment]
➤ Lauren Zugai commented:

[~jgruen@mozilla.com] is it possible to change whatever is causing "Password Reset - 1" to be read in SurveyGizmo? The "1" feels awkward. If we go the `aria-label` route to read "Firefox accounts optional user survey", "Password reset survey form" would be great (or at least just remove the "1" if possible).

**Comment by data-sync-user** [comment]
➤ John Gruen commented:

Yes, can you file that as separate and assign it to me? I can fix it in the template.

For minimize v close, legal actually requested we switch to close which is why i'm pushing for it.

**Comment by data-sync-user** [comment]
➤ Lauren Zugai commented:

Filed as a separate issue here: FXA-2228

As discussed in sprint planning, we've made a spin-off issue for changing the minimize functionality to close, FXA-2226, which we'll add an `aria-label` to there.

That will change the scope of this issue since the bullet points no longer apply to the button. All this issue needs now is the aria-label on the container so I'm changing it from a 3 to a 1.
